packages/abi-coder/src/abi-coder.ts

```diff
@@ -82,7 +83,9 @@ export default class AbiCoder {
         obj[component.name] = this.getCoder(component);
         return obj;
       }, {});
-      return new EnumCoder(enumMatch.name, coders);
+      return param.type === 'enum Option'
```

Comment:
I was trying to think of a better way to de-nest this but couldn't.  My gut tells me we should refactor `getCoder` but probably not a blocker for this pr